### ISSUE TITLE
Example Project

### ISSUE THREAD

**Issue by newtack** [awslabs/dynamodb-data-mapper-js#76]
I like the library and the documentation is OK, but I am really missing examples and recipes.

Right now I am stuck on how to ensure that on a put that it fails if the ID already exists and allows me to retry with another randomly generated id. I assume that I should use some expression, but can't find an example or recipe. 

I'd love to see recipes or sample project that show examples of how this code looks like, this would be very helpful.

**Comment by jeskew** [comment]
That totally makes sense. I could set up a simple "usage examples" page at the top level of the repository documentation.

**Comment by LucasBadico** [comment]
I could help with that. I'm building a repository pattern with it, based on DDD directly from the book. My BDD cases could be use as recepies, besides in the majority be using just a little of the resourses form this project.

**Comment by digitaldivvy** [comment]
Documentation is preliminary - impossible so far to use in a real project.

Is there a working example for basic objects, embedded objects etc. with different types across basic CRUD and Queries with conditional expressions?

**Comment by LucasBadico** [comment]
I having using for the past months @digitaldivvy .

**Comment by acheronfail** [comment]
Yep - this library is great. We currently have a multiple services in production using it, and it's working great for us.

**Comment by digitaldivvy** [comment]
ditto - awesome work aws!
